### PR TITLE
Support Multiple Base Layers in BaseMap

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This PR adds support for multiple base layers. Configuration is done by allowing `baseLayer` to accept an array of strings instead of just a string.

A new prop `baseLayerNames` is added to allow the control to display pretty names. An alternative might be to use internationalization with the url as a key, but I think it's easier to push that responsibility up a level? Curious for others' thoughts.
Screen Shot 2022-12-02 at 11 33 10 AM

### CODE REVIEW

**Review by demory — APPROVED** [review]

**Comment by demory** [comment]
I approved this but will add one suggestion: maybe add a second item to storybook that demos both the base layer selector and one or more overlays together?

File: packages/base-map/src/index.story.tsx [hunk 1]
```diff
@@ -179,3 +179,28 @@ export const withOptionalControls = () => (
...
 withOptionalControls.decorators = [withMap()];
+
+export const withMultipleBaseLayers = () => {
+  const [mapTilerKey, setMapTilerKey] = useState("");
+  return (
+    <Styled.StoryMapContainer>
+      <input
+        type="text"
+        value={mapTilerKey}
+        placeholder="MapTiler API Key"
```
Comment (binh-dam-ibigroup) [inline packages/base-map/src/index.story.tsx:190]: Sort props (multiple instances).

File: packages/base-map/src/index.tsx [hunk 2]
```diff
@@ -23,7 +23,9 @@ import MarkerWithPopup from "./MarkerWithPopup";
  */
 type Props = React.ComponentPropsWithoutRef<React.ElementType> & {
   /** A URL pointing to the vector tile specification which should be used as the main map.  */
```
Comment (binh-dam-ibigroup) [inline packages/base-map/src/index.tsx:25]: Update the JSDoc for arrays of URLs.

File: packages/base-map/src/index.tsx [hunk 3]
```diff
@@ -102,6 +105,9 @@ const BaseMap = ({
   const [hiddenLayers, setHiddenLayers] = useState(
     toggleableLayers.filter(layer => !layer?.visible).map(layer => layer.id)
   );
+  const [activeBaseLayer, setActiveBaseLayer] = useState(
+    typeof baseLayer === "object" ? baseLayer?.[0] : baseLayer
```
Comment (binh-dam-ibigroup) [inline packages/base-map/src/index.tsx:109]: From the type definitions, there is an assumption that `baseLayer` can be `null`. Is that correct? Also, `typeof null === 'object'` is true. So, a null check is probably needed on lines 142 and 154. Alternately, create a variable for whether `baseLayer` is a non-null array and use that throughout this component.

File: packages/base-map/src/index.tsx [hunk 4]
```diff
@@ -143,6 +150,24 @@ const BaseMap = ({
           <ul
             className={`layers-list ${fakeMobileHover && "fake-mobile-hover"}`}
           >
+            {typeof baseLayer === "object" &&
+              baseLayer.map((layer: string, index: number) => {
+                return (
+                  <li key={index}>
+                    {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+                    <label>
```
Comment (binh-dam-ibigroup) [inline packages/base-map/src/index.tsx:158]: Go ahead and add the `htmlFor` prop to the label, even if it already contains the input control, and delete the a11y exception.

File: packages/base-map/src/index.tsx [hunk 5]
```diff
@@ -143,6 +150,24 @@ const BaseMap = ({
...
+                      <input
+                        type="radio"
+                        checked={activeBaseLayer === layer}
```
Comment (binh-dam-ibigroup) [inline packages/base-map/src/index.tsx:161]: Sort props.

File: packages/base-map/src/index.tsx (hunk 4 above)
Comment (binh-dam-ibigroup) [reply packages/base-map/src/index.tsx:158]: (Might do that for the checkbox labels below too)

File: packages/base-map/src/index.tsx [hunk 6]
```diff
@@ -143,6 +150,24 @@ const BaseMap = ({
           <ul
             className={`layers-list ${fakeMobileHover && "fake-mobile-hover"}`}
           >
+            {typeof baseLayer === "object" &&
+              baseLayer.map((layer: string, index: number) => {
+                return (
+                  <li key={index}>
```
Comment (binh-dam-ibigroup) [inline packages/base-map/src/index.tsx:156]: We can probably use `layer` as the key instead of the index.

File: packages/base-map/src/index.tsx [hunk 7]
```diff
@@ -23,7 +23,9 @@ import MarkerWithPopup from "./MarkerWithPopup";
  */
 type Props = React.ComponentPropsWithoutRef<React.ElementType> & {
   /** A URL pointing to the vector tile specification which should be used as the main map.  */
-  baseLayer?: string;
+  baseLayer?: string | string[];
+  /** A list of names to match onto the base layers */
```
Comment (binh-dam-ibigroup) [inline packages/base-map/src/index.tsx:27]: Also say in the JSDoc that `baseLayerNames` is used only if multiple entries are defined for `baseLayer`.

**Review by binh-dam-ibigroup — APPROVED** [review]
Approving subject to some code cleanup as I indicated and subject to @demory's suggestion to add a storybook example showing multiple base layers with overlays.

File: packages/base-map/src/index.tsx (hunk 3 above)
Comment (miles-grant-ibigroup) [reply packages/base-map/src/index.tsx:109]: Very good catch thank you! Forgot `null` is an object how we love javascript

File: packages/base-map/src/index.tsx (hunk 4 above)
Comment (miles-grant-ibigroup) [reply packages/base-map/src/index.tsx:158]: Why are we changing this now? We've been using this pattern everywhere in the project so far

File: packages/base-map/src/index.tsx (hunk 6 above)
Comment (miles-grant-ibigroup) [reply packages/base-map/src/index.tsx:156]: I had this but key `0` caused problems

**Pull request merged**